geniza/corpus/management/commands/import_data.py

```diff
@@ -336,3 +372,153 @@ def log_creation(self, *objects):
                 object_repr=str(obj),
                 change_message=self.logentry_message,
                 action_flag=ADDITION)
+
+    def get_user(self, name, pgpid=None):
+        """Find a user account based on a provided name, using a simple cache.
+
+        If not found, tries to use first/last initials for lookup. If all else
+        fails, use the generic team account (TEAM_USERNAME).
+        """
+
+        # check the cache first
+        user = self.user_lookup.get(name)
+        if user:
+            logging.debug(
+                f"Using cached user {user} for {name} on PGPID {pgpid}")
+            return user
+
+        # person with given name(s) and last name – case-insensitive lookup
+        if " " in name:
+            given_names, last_name = [sname.strip(punctuation) for sname in
+                                      name.rsplit(" ", 1)]
+            try:
+                user = User.objects.get(first_name__iexact=given_names,
+                                        last_name__iexact=last_name)
+            except User.DoesNotExist:
+                pass
+
+        # initials; use first & last to do lookup
+        elif name:
+            first_i, last_i = name[0], name[-1]
+            try:
+                user = User.objects.get(first_name__startswith=first_i,
+                                        last_name__startswith=last_i)
+            except User.DoesNotExist:
+                pass
+
+        # if we didn't get anyone through either method, warn and use team user
+        if not user:
+            logging.warning(
+                f"Couldn't find user {name} on PGPID {pgpid}; using {self.team_user}")
```

Comment:
That's because they're already in the database!

Reply:
They might be in QA, but they aren't in prod yet because this app doesn't exist there, and they weren't in my local dev database. I could see using `createcasuser` to make sure the user accounts exist rather than putting them in the fixture (which I agree seems a little weird), but it needs to be documented in deploy notes as a step to run (with any usernames that are required for data import) _before_ running the data import.

Reply:
hmm, what do you think about creating them with `createcasuser` via `call_command` in this script? presumably it will do nothing if they already exist

Reply:
even better! It will update the accounts from CAS if information has changed, but it won't cause any problems if they already exist. (You should go ahead and pass the admin flag also)